CMakeLists.txt

```diff
@@ -15,9 +15,13 @@ message(STATUS "CMAKE_SYSTEM_PROCESSOR = ${CMAKE_SYSTEM_PROCESSOR}")
 message(STATUS "CMAKE_CXX_COMPILER_ID = ${CMAKE_C_COMPILER_ID}")
 
 if("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86_64" OR  "${CMAKE_SYSTEM_PROCESSOR}" STREQUAL AMD64)
-   if (CMAKE_C_COMPILER_ID STREQUAL "GNU" OR CMAKE_C_COMPILER_ID STREQUAL "Clang")
+   if (CMAKE_C_COMPILER_ID STREQUAL "GNU" OR CMAKE_C_COMPILER_ID STREQUAL "Clang" OR CMAKE_C_COMPILER_ID STREQUAL "AppleClang")
     message(STATUS "compiling with -march=native")
     target_compile_options(despacer PRIVATE -march=native)
+  elseif(CMAKE_C_COMPILER_ID STREQUAL "MSVC")
+    message(STATUS "Assuming AVX2, SSE4.2, SSE4.1 support on MSVC")
+    target_compile_options(despacer PRIVATE /arch:AVX2)
+    target_compile_definitions(despacer PRIVATE __SSE4_1__ __SSE4_2__ __AVX2__)
```

Comment:
As per the Microsoft documentation:

```The __AVX__ preprocessor symbol is defined when the /arch:AVX, /arch:AVX2 or /arch:AVX512 compiler option is specified. The __AVX2__ preprocessor symbol is defined when the /arch:AVX2 or /arch:AVX512 compiler option is specified. ```

Reply:
I'd rather that the macros starting with `_` not be manually set. Let the compiler do it.